scene/2d/tile_map.cpp

```diff
@@ -1831,6 +1951,8 @@ TileMap::TileMap() {
 	collision_parent = NULL;
 	navigation = NULL;
 	y_sort_mode = false;
+	compatibility_mode = true; //To not break compatibility with godot 3.0-3.1
```

Comment:
I'd say let's make it false by default, and we can document this change in the release notes. So people will use the non-compatibility mode when creating new projects in 3.2, but they'll have to enable this option when porting older projects which need this feature (most don't need it, so they also won't notice the change).